packages/react-storage/src/components/StorageBrowser/controls/hooks/__tests__/useDataRefresh.spec.ts

```diff
@@ -0,0 +1,37 @@
+import { renderHook } from '@testing-library/react';
+import { DataRefreshProps } from '../../../composables/DataRefresh';
+import { useControlsContext } from '../../../controls/context';
+import { useDataRefresh } from '../useDataRefresh';
+
+jest.mock('../../../controls/context');
+
+describe('useDataRefresh', () => {
+  const data = {
+    dataRefreshLabel: 'add-folder-label',
+    isDataRefreshDisabled: false,
+  };
+
+  const mockUseControlsContext = jest.mocked(useControlsContext);
+
+  beforeEach(() => {
+    mockUseControlsContext.mockReturnValue({
+      data,
+      onRefresh: jest.fn(),
+    });
+  });
+
+  afterEach(() => {
+    mockUseControlsContext.mockReset();
+  });
+
+  it('returns DataRefresh props', () => {
+    const { result } = renderHook(() => useDataRefresh());
+
+    const expected: DataRefreshProps = {
+      isDisabled: data.isDataRefreshDisabled,
+      onRefresh: expect.any(Function),
```

Comment:
we could consider using mock `onRefresh` from line 19, possibly expect `onRefresh: mockOnRefresh` instead of any function. this applies to a lot of the other tests, but i don't think it's critical.

Reply:
Yeah, I had considered doing this but we'd have to pull them out into `const mockOnRefresh = jest.fn()` just to assert something that's more or less trivial in the end. So just opted to keep it marginally more concise